File: blas/src/saxpy.f90
program blas_saxpy
  implicit none

  integer, parameter :: N = 10
  
  integer :: i
  real    :: alpha, x(N), y(N)

  ! initialize
  alpha = 10
  do i=1,N
     x(i) = i
     y(i) = 2*i
  end do

  ! y = alpha * x + y
  CALL SAXPY( N, alpha, x, 1, y, 1 )

  ! check
  do i=1,N
     print *, y(i), " should equal ", alpha*i + 2*i
  end do

end program blas_saxpy

File: blas/src/saxpy.cpp
#include <iostream>

#include "/System/Library/Frameworks/Accelerate.framework/Frameworks/vecLib.framework/Headers/cblas.h"

int main() {

  const int N = 10;
  float x[N], y[N];
  float alpha = 10;

  // initialize
  for ( int i=0; i<N; i++ ) {
    x[i] = i;
    y[i] = 2*i;
  }

  // y = alpha * x + y
  cblas_saxpy( N, alpha, x, 1, y, 1 );

  for ( int i=0; i<N; i++ )
    std::cout << y[i] << " should equal " << alpha*i + 2*i << std::endl;
  
  return 0;
  
}
